internal-api/src/main/java/datadog/trace/api/Config.java

```diff
@@ -1034,9 +1040,15 @@ private Config(final ConfigProvider configProvider, final InstrumenterConfig ins
 
     clockSyncPeriod = configProvider.getInteger(CLOCK_SYNC_PERIOD, DEFAULT_CLOCK_SYNC_PERIOD);
 
-    logsInjectionEnabled =
-        configProvider.getBoolean(
-            LOGS_INJECTION_ENABLED, DEFAULT_LOGS_INJECTION_ENABLED, LOGS_INJECTION);
+    if (experimentalFeaturesEnabled.contains(
+        propertyNameToEnvironmentVariableName(LOGS_INJECTION))) {
+      logsInjectionEnabled =
+          configProvider.getBoolean(
+              LOGS_INJECTION_ENABLED, DEFAULT_LOGS_INJECTION_ENABLED, LOGS_INJECTION);
+    } else {
+      logsInjectionEnabled =
+          configProvider.getBoolean(LOGS_INJECTION_ENABLED, true, LOGS_INJECTION);
+    }
```

Comment:
Suggestion: keep `DEFAULT_LOGS_INJECTION_ENABLED` as `true` for now and use it for the non-breaking-changes case, then use `false` for the breaking-changes case.
```suggestion
    if (experimentalFeaturesEnabled.contains(
        propertyNameToEnvironmentVariableName(LOGS_INJECTION))) {
      logsInjectionEnabled =
          configProvider.getBoolean(
              LOGS_INJECTION_ENABLED, false, LOGS_INJECTION);
    } else {
      logsInjectionEnabled =
          configProvider.getBoolean(LOGS_INJECTION_ENABLED, DEFAULT_LOGS_INJECTION_ENABLED, LOGS_INJECTION);
    }
```
In other words postpone updating `DEFAULT_LOGS_INJECTION_ENABLED` value until we prepare the codebase for 2.x